### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by colinoflynn**
This is my updates to the documentation to reflect my more major changes. There are a number of TODOs in there still, will continue to update but wanted to have this PR active for easier commenting from everyone.

You can see the mdbook version of these at https://newaetech.github.io/sonata-system/ which is easier to review.

### CODE REVIEW

**Comment by colinoflynn** [comment]
Also - I can squash all these commits into a single one as the commit messages were mostly in-progress work. I didn't want to do that before reviewing though as it may be helpful not having it all squashed to see the specific changes.

**Review by HU90m — DISMISSED** [review]

File: doc/guide/fpga-update.md
Comment (colinoflynn) [reply doc/guide/fpga-update.md:32]: Ah - I was looking at the old revision PCB when I wrote this.

Comment (colinoflynn) [reply doc/guide/fpga-update.md]: Fixed now, thanks for catching that!

File: doc/dev/developing.md
Comment (marnovandermaas) [inline doc/dev/developing.md:52]: We can remove these bullet points as they are remnants from the old documentation

File: doc/dev/fpga-development.md
Comment (marnovandermaas) [inline doc/dev/fpga-development.md:1]:
```suggestion
# FPGA development

## Dependencies
```

File: doc/guide/fpga-update.md
Comment (marnovandermaas) [inline doc/guide/fpga-update.md:138]: This is repeated from the FPGA programming section, maybe just leave it there for now?

**Review by marnovandermaas — COMMENTED** [review]
Just a few comments from my side. Once those are resolved I'm happy to merge this and iterate on it after that.

Also, we need to make sure to squash this before merging.

File: doc/guide/toolchain-setup.md
Comment (HU90m) [inline doc/guide/toolchain-setup.md:70]: nit:
```suggestion
> ```sh
> wget https://github.com/llvm/llvm-project/commit/4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
> git am < 4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
> ```
```

**Comment by colinoflynn** [comment]
Thanks all - I'll get these fixed up for tonight my time (aka probably tomorrow for you) & squash it down to a nice single commit once I do. Will flag it back once I think it's ready!

File: doc/guide/fpga-update.md
Comment (colinoflynn) [reply doc/guide/fpga-update.md:138]: I made a short note you can use this feature for debug & linked to the fpga programming section.

File: doc/dev/developing.md
Comment (marnovandermaas) [reply doc/dev/developing.md]: Thanks Colin!

File: doc/guide/fpga-update.md
Comment (marnovandermaas) [reply doc/guide/fpga-update.md:138]: Ok, sounds good.

**Comment by colinoflynn** [comment]
It has been done. https://newaetech.github.io/sonata-system/ was built successfully from our repo, so can use that for checking any changes.

**Review by marnovandermaas — APPROVED** [review]
This looks like a good start to me. Thanks for starting this documentation rewrite. Let's get this into the main repository and iterate from there.

**Pull request merged**
